lib/iris/cube.py

```diff
@@ -1632,7 +1638,38 @@ def dtype(self):
 
     @dtype.setter
     def dtype(self, dtype):
-        self._dtype = dtype
+        if dtype != self.dtype:
+            if dtype is not None:
+                if not self.has_lazy_data():
+                    emsg = 'Cube does not have lazy data, cannot set dtype.'
+                    raise ValueError(emsg)
+                dtype = np.dtype(dtype)
+                if dtype.kind != 'i':
+                    emsg = ('Can only cast lazy data to integral dtype, '
+                            'got {!r}.')
```

Comment:
Hmmm good point @pp-mo ... See this [concatenate](https://github.com/SciTools/iris/pull/2433/files#diff-5fa4649482766af96ae11aead017e334R340) change ... is there a generic numpy kind to catch all integral kinds instead of `i` and `u` ?